system-test/src/systemTest/java/org/apache/mesos/elasticsearch/systemtest/RunAsJarSystemTest.java

```diff
@@ -0,0 +1,150 @@
+package org.apache.mesos.elasticsearch.systemtest;
+
+import com.containersol.minimesos.MesosCluster;
+import com.containersol.minimesos.container.AbstractContainer;
+import com.containersol.minimesos.mesos.*;
+import com.github.dockerjava.api.DockerClient;
+import com.github.dockerjava.api.command.CreateContainerCmd;
+import com.github.dockerjava.api.model.ExposedPort;
+import com.github.dockerjava.api.model.Link;
+import com.github.dockerjava.api.model.Ports;
+import org.apache.mesos.elasticsearch.common.cli.ElasticsearchCLIParameter;
+import org.apache.mesos.elasticsearch.common.cli.ZookeeperCLIParameter;
+import org.apache.mesos.elasticsearch.scheduler.Configuration;
+import org.apache.mesos.elasticsearch.systemtest.callbacks.ElasticsearchNodesResponse;
+import org.apache.mesos.elasticsearch.systemtest.util.DockerUtil;
+import org.junit.*;
+import org.junit.rules.TestWatcher;
+import org.junit.runner.Description;
+
+import java.io.IOException;
+import java.security.SecureRandom;
+import java.util.stream.IntStream;
+
+import static org.junit.Assert.assertTrue;
+
+/**
+ * A system test to ensure that the framework can run as a JAR, not using docker.
+ */
+@SuppressWarnings({"PMD.TooManyMethods", "PMD.AvoidUsingHardCodedIP"})
+public class RunAsJarSystemTest {
+    protected static final org.apache.mesos.elasticsearch.systemtest.Configuration TEST_CONFIG = new org.apache.mesos.elasticsearch.systemtest.Configuration();
+    private static final int NUMBER_OF_TEST_TASKS = 1;
+
+    // Need full control over the cluster, so need to do all the lifecycle stuff.
+    private MesosCluster cluster;
+    private static DockerClient dockerClient = DockerClientFactory.build();
+    private JarScheduler scheduler;
+
+    @Before
+    public void before() {
+        ClusterArchitecture.Builder builder = new ClusterArchitecture.Builder()
+                .withZooKeeper()
+                .withMaster()
+                .withSlave() // Have to have a slave before we build. Bit of a minimesos bug. This offer wont get accepted.
+                .withContainer(zkContainer -> new JarScheduler(dockerClient, zkContainer), ClusterContainers.Filter.zooKeeper());
+        scheduler = (JarScheduler) builder.build().getClusterContainers().getOne(container -> container instanceof JarScheduler).get();
+        IntStream.range(0, NUMBER_OF_TEST_TASKS).forEach(dummy ->
+            builder.withSlave(zooKeeper -> new MesosSlaveWithSchedulerLink(dockerClient, zooKeeper, scheduler))
+        );
+        cluster = new MesosCluster(builder.build());
+
+        cluster.start();
+    }
+
+    @After
```

Comment:
Maybe sort the `@BeforeClass`, `@Before`, `@Rule`, `@After`, `@AfterClass` in the order they're being executed?